libraries/Adapters/Microsoft.Bot.Builder.Adapters.Slack/SlackClientWrapper.cs

```diff
@@ -0,0 +1,554 @@
+        /// </summary>
+        /// <returns>A <see cref="DirectMessageConversationListResponse"/> representing the response to listing the messages.</returns>
+        public virtual async Task<DirectMessageConversationListResponse> GetDirectMessageListAsync()
+        {
+            return await _api.GetDirectMessageListAsync().ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetFileInfoAsync method.
+        /// </summary>
+        /// <param name="fileId">The id of the file to retrieve the info from.</param>
+        /// <param name="page">The page number.</param>
+        /// <param name="count">The requested count.</param>
+        /// <returns>A <see cref="FileInfoResponse"/> representing the response.</returns>
+        public virtual async Task<FileInfoResponse> GetFileInfoAsync(string fileId, int? page = null, int? count = null)
+        {
+            return await _api.GetFileInfoAsync(fileId, page, count).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetFilesAsync method.
+        /// </summary>
+        /// <param name="userId">The user id.</param>
+        /// <param name="from">The initial date.</param>
+        /// <param name="to">The final date.</param>
+        /// <param name="count">The requested count.</param>
+        /// <param name="page">The page number.</param>
+        /// <param name="types">The type of files to retrieve.</param>
+        /// <returns>A <see cref="FileListResponse"/> representing the response.</returns>
+        public virtual async Task<FileListResponse> GetFilesAsync(string userId = null, DateTime? from = null, DateTime? to = null, int? count = null, int? page = null, FileTypes types = FileTypes.all)
+        {
+            return await _api.GetFilesAsync(userId, from, to, count, page, types).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetGroupHistoryAsync method.
+        /// </summary>
+        /// <param name="groupInfo">Info about the channel.</param>
+        /// <param name="latest">The date of the latest.</param>
+        /// <param name="oldest">The date of the oldest.</param>
+        /// <param name="count">The requested count.</param>
+        /// <returns>A <see cref="GroupMessageHistory"/> representing the asynchronous operation.</returns>
+        public virtual async Task<GroupMessageHistory> GetGroupHistoryAsync(Channel groupInfo, DateTime? latest = null, DateTime? oldest = null, int? count = null)
+        {
+            return await _api.GetGroupHistoryAsync(groupInfo, latest, oldest, count).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Wraps Slack API's GetGroupsListAsync method.
+        /// </summary>
+        /// <param name="excludeArchived">Flag setting if archived groups are to be excluded.</param>
+        /// <returns>A <see cref="GroupListResponse"/> representing the asynchronous operation.</returns>
```

Comment:
The return here is the requested list of groups.

Reply:
my bad